src/spatial/nearest_neighbor.cpp

```diff
@@ -1,8 +1,59 @@
 
+#include "igraph_operators.h"
+#include "igraph_spatial.h"
 #include "igraph_datatype.h"
+#include "igraph_matrix.h"
 #include "igraph_error.h"
 #include "igraph_types.h"
+#include "igraph_vector.h"
+#include "nanoflann.hpp"
```

Comment:
Where do you keep this header now? In the igraph source tree, or are you relying on a system-wide version? I think Arch doesn't have it as a separate package: https://repology.org/project/nanoflann/versions